pkg/component/extensions/operatingsystemconfig/nodeinit/nodeinit.go

```diff
@@ -117,6 +120,24 @@ WantedBy=multi-user.target`),
 	}
 	nodeInitFiles = append(nodeInitFiles, nodeAgentFiles...)
 
+	// Create gardener user and ensure SSHD is enabled when SSH access for the nodes is enabled.
+	// This simplifies debugging errors where gardener-node-agent might not be bootstrapped successfully.
+	if ctx.SSHAccessEnabled {
+		userUnits, userFiles, err := gardeneruser.New().Config(ctx)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed computing gardener-user config: %w", err)
+		}
+		nodeInitUnits = append(nodeInitUnits, userUnits...)
+		nodeInitFiles = append(nodeInitFiles, userFiles...)
+
+		sshdUnits, sshdFiles, err := sshdensurer.New().Config(ctx)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed computing sshd-ensurer config: %w", err)
+		}
+		nodeInitUnits = append(nodeInitUnits, sshdUnits...)
+		nodeInitFiles = append(nodeInitFiles, sshdFiles...)
+	}
```

Comment:
I would not include this here - this file should be solely used for artefacts related to `gardener-node-init`. https://github.com/gardener/gardener/blob/5fc5009be3f3dc09ca08e991da786a3a4a9d0efa/pkg/component/extensions/operatingsystemconfig/operatingsystemconfig.go#L669-L674 is the proper place to put it (where the OSC is assembled).